src/ca/idcert.rs

```diff
@@ -178,23 +182,30 @@ impl IdCert {
     /// by the provided `issuer` certificate.
     ///
     /// Note that this does _not_ check the CRL.
-    pub fn validate_ee(&self, issuer_key: &PublicKey) -> Result<(), ValidationError> {
+    pub fn validate_ee(
+        &self, issuer_key: &PublicKey,
+    ) -> Result<(), ValidationError> {
         self.validate_ee_at(issuer_key, Time::now())
     }
 
-    pub fn validate_ee_at(&self, issuer_key: &PublicKey, now: Time) -> Result<(), ValidationError> {
-        self.validate_basics(now)?;
-        self.validate_aki(issuer_key)?;
+    pub fn validate_ee_at(
+        &self, issuer_key: &PublicKey, now: Time,
+    ) -> Result<(), ValidationError> {
+        self.inspect_basics()?;
+        self.verify_validity(now)?;
+        self.verify_issuer_key(issuer_key)?;
 
         // Basic Constraints: Must not be a CA cert.
         if let Some(basic_ca) = self.basic_ca {
             if basic_ca {
-                return Err(ValidationError);
+                return Err(VerificationError::new(
+                    "Basic Constraints extions in end entity cert"
```

Comment:
Here you say "end entity" but in the if test and comment above you refer to "CA", are they both right or is one wrong?